py/dml/c_backend.py

```diff
@@ -1992,19 +1992,31 @@ def init_trait_vtables_for_node(node, param_values):
                             f'{TPtr(array_type).declaration(var)} = malloc('
                             f'sizeof(*{var}));\n')
                         if deep_const(t):
-                            k = TArray(t, mkIntegerLiteral(node.site, 1)).declaration("")
+                            k = (TArray(t, mkIntegerLiteral(node.site, 1))
+                                 .declaration(""))
+                            target = f'(void *)&(*{var}){param_indices}'
+                            source = f'({k}) {{ {value} }}'
+                            size = f'sizeof({t.declaration("")})'
                             param_init.append(
-                                f' memcpy(&(*{var}){param_indices},'
```

Comment:
This causes a GCC error if `t` is const-qualified, as we drop the const-qualifier of the pointer basetype by doing this -- you need to explicitly cast to `void *`. Extended the `T_trait.dml` test to cover this.